File: tests/integration/fixtures/simple_service.cs
/// <summary>
/// Simple payment processing service - C# version
/// </summary>
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PaymentService
{
    public class User
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public double Balance { get; set; }

        public User(string id, string name, string email, double balance)
        {
            Id = id;
            Name = name;
            Email = email;
            Balance = balance;
        }
    }

    public class Transaction
    {
        public string TransactionId { get; set; }
        public string Status { get; set; }
        public double Amount { get; set; }
        public string UserId { get; set; }

        public Transaction(string transactionId, string status, double amount, string userId)
        {
            TransactionId = transactionId;
            Status = status;
            Amount = amount;
            UserId = userId;
        }
    }

    public class PaymentProcessor
    {
        private Dictionary<string, User> users;
        private List<Transaction> transactions;

        public PaymentProcessor()
        {
            users = new Dictionary<string, User>();
            transactions = new List<Transaction>();
        }

        public User GetUser(string userId)
        {
            return users.ContainsKey(userId) ? users[userId] : null;
        }

        public void AddUser(User user)
        {
            users[user.Id] = user;
        }

        public Transaction ProcessPayment(string userId, double amount)
        {
            var user = GetUser(userId);

            if (user == null)
            {
                throw new Exception($"User not found: {userId}");
            }

            if (user.Balance < amount)
            {
                throw new Exception($"Insufficient balance: {user.Balance} < {amount}");
            }

            // Update balance
            user.Balance -= amount;

            // Create transaction
            var transaction = new Transaction(
                $"tx_{transactions.Count + 1}",
                "completed",
                amount,
                userId
            );

            transactions.Add(transaction);
            return transaction;
        }

        public double GetBalance(string userId)
        {
            var user = GetUser(userId);
            return user != null ? user.Balance : 0.0;
        }

        public List<Transaction> ListTransactions(string userId)
        {
            return transactions.Where(t => t.UserId == userId).ToList();
        }
    }

    public static class PaymentUtils
    {
        public static double CalculateFee(double amount)
        {
            if (amount < 10.0)
            {
                return 0.5;
            }
            else if (amount < 100.0)
            {
                return amount * 0.02;
            }
            else {
                return amount * 0.01;
            }
        }

        public static async Task<bool> AsyncValidatePayment(string userId, double amount)
        {
            // Simulate async validation
            await Task.Delay(0);
            return amount > 0;
        }
    }
}

File: tests/integration/fixtures/simple_service.rs
/// Simple payment processing service - Rust version
use std::collections::HashMap;

#[derive(Debug, Clone)]
pub struct User {
    pub id: String,
    pub name: String,
    pub email: String,
    pub balance: f64,
}

#[derive(Debug, Clone)]
pub struct Transaction {
    pub transaction_id: String,
    pub status: String,
    pub amount: f64,
    pub user_id: String,
}

pub struct PaymentProcessor {
    users: HashMap<String, User>,
    transactions: Vec<Transaction>,
}

impl PaymentProcessor {
    pub fn new() -> Self {
        PaymentProcessor {
            users: HashMap::new(),
            transactions: Vec::new(),
        }
    }

    pub fn get_user(&self, user_id: &str) -> Option<&User> {
        self.users.get(user_id)
    }

    pub fn add_user(&mut self, user: User) {
        self.users.insert(user.id.clone(), user);
    }

    pub fn process_payment(&mut self, user_id: &str, amount: f64) -> Result<Transaction, String> {
        let user = self.users.get_mut(user_id)
            .ok_or_else(|| format!("User not found: {}", user_id))?;

        if user.balance < amount {
            return Err(format!("Insufficient balance: {} < {}", user.balance, amount));
        }

        // Update balance
        user.balance -= amount;

        // Create transaction
        let transaction = Transaction {
            transaction_id: format!("tx_{}", self.transactions.len() + 1),
            status: "completed".to_string(),
            amount,
            user_id: user_id.to_string(),
        };

        self.transactions.push(transaction.clone());
        Ok(transaction)
    }

    pub fn get_balance(&self, user_id: &str) -> f64 {
        self.get_user(user_id)
            .map(|u| u.balance)
            .unwrap_or(0.0)
    }

    pub fn list_transactions(&self, user_id: &str) -> Vec<Transaction> {
        self.transactions
            .iter()
            .filter(|t| t.user_id == user_id)
            .cloned()
            .collect()
    }
}

pub fn calculate_fee(amount: f64) -> f64 {
    if amount < 10.0 {
        0.5
    } else if amount < 100.0 {
        amount * 0.02
    } else {
        amount * 0.01
    }
}

pub async fn async_validate_payment(user_id: &str, amount: f64) -> bool {
    // Simulate async validation
    amount > 0.0
}
